api/app/controllers/transaction.controller.ts

```diff
@@ -78,55 +80,85 @@ exports.create = async (req, res) => {
     );
     return res.send(transactionResult);
   } catch (error) {
-    transaction.transactionStatus = TRANSACTION_STATUS.FAILURE;
+    const message =
+      error instanceof Error
+        ? error.message
+        : "Some error occurred while creating the Transaction.";
+    transaction.transactionStatus = TRANSACTION_STATUS.FAILED;
     Transaction.create(transaction);
     res.status(500).send({
-      message:
-        err.message || "Some error occurred while creating the Transaction.",
+      message,
     });
   }
-};
+}
 
-exports.findAll = async (req, res) => {
+export async function findAll(req: Request, res: Response) {
   const { userId, status, type } = req.query;
   // validate fields
   if (!userId) {
     res.status(400).send({
       message: `Missing required fields: userId`,
     });
     return;
-  } else if (isNaN(userId)) {
+  } else if (isNaN(Number(userId))) {
```

Comment:
This Number conversion is redundant imo. We are checking if `userId` is a number but converting the `userId` to a number before checking if its a number???!

Reply:
The Number constructor is used here because the type of `userId` could be a `string | string[] | ParsedQs | ParsedQs[]` and `isNaN` expects a number type as its argument